### PR TITLE
Use Delegate model instead of manifest file

### PR DESCRIPTION
**Pull request by bion**

### CODE REVIEW

**Comment by bion** [comment]
@dbalatero @scottmacphersonmusic @andyjac re-view

File: app/lib/send_messages.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class SendMessages
-  def self.to_delegates(message, delegates)
-    delegates.each do |delegate|
+  def self.to_delegates(message)
+    message.delegates.each do |delegate|
```
Comment (dbalatero) [inline app/lib/send_messages.rb]: tight

File: app/views/delegates/index.html.rb [hunk 2]
```diff
@@ -80,26 +80,31 @@ def message_fields(f)
   end
 
   def delegate_inputs(f)
-    delegates.each do |delegate|
-      delegate_switch(f, delegate)
+    delegates.each do |d|
```
Comment (dbalatero) [inline app/views/delegates/index.html.rb]: :rotating_light: `s/d/delegate/g`

File: config/routes.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 Rails.application.routes.draw do
-  root 'delegates#index'
+  root 'delegates#index', state: 'wa'
```
Comment (dbalatero) [inline config/routes.rb]: I'd like to force all states to be `upcase` eg `WA`

File: app/models/delegate.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class Delegate < ActiveRecord::Base
+  validates :state,
```
Comment (dbalatero) [inline app/models/delegate.rb]:
```ruby
before_validation :ensure_state_is_upcased

def ensure_state_is_upcased
  return unless state.present?

  self.state = self.state.upcase
end```

Comment (dbalatero) [inline app/models/delegate.rb]: If you like, extract this from `Message` and `Delegate` into a module?

File: spec/support/vcr.rb [hunk 5]
```diff
@@ -3,4 +3,5 @@
 VCR.configure do |config|
   config.cassette_library_dir = "spec/fixtures/vcr_cassettes"
   config.hook_into :webmock
+  config.ignore_hosts '127.0.0.1', 'localhost'
```
Comment (dbalatero) [inline spec/support/vcr.rb]: hmm, this is needed to eh?

File: spec/support/webmock.rb [hunk 6]
```diff
@@ -1,3 +1,5 @@
 require 'webmock/rspec'
 
-WebMock.disable_net_connect!
+WebMock.disable_net_connect! \
+  allow_localhost: true,
+  allow: 'http://127.0.0.1'
```
Comment (dbalatero) [inline spec/support/webmock.rb]: Are both of these needed?

**Comment by dbalatero** [comment]
@bion comments left

File: spec/support/vcr.rb (hunk 5 above)
Comment (bion) [inline spec/support/vcr.rb]: evidenlty

File: spec/support/webmock.rb (hunk 6 above)
Comment (bion) [inline spec/support/webmock.rb]: yeah

**Pull request merged**